pkg/spotio/spotio.go

```diff
@@ -0,0 +1,127 @@
+package spotio
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"golang.org/x/oauth2"
+)
+
+const (
+	spotIOBaseURL  = "https://api.spotinst.io"
+	launchSpecPath = "/ocean/aws/k8s/launchSpec"
+)
+
+// OceanTag is a tag defined for an ocean cluster.
+type OceanTag struct {
+	Key   string `json:"tagKey"`
+	Value string `json:"tagValue"`
+}
+
+// OceanLaunchSpec is the launcSpec defined for an ocean cluster.
+type OceanLaunchSpec struct {
+	ID       string     `json:"id"`
+	OceanID  string     `json:"oceanId"`
+	ImageID  string     `json:"imageId"`
+	UserData string     `json:"userData"`
+	Tags     []OceanTag `json:"tags"`
+}
+
+// API is an interface describing the Spot.io HTTP API.
+type API interface {
+	ListLaunchSpecs(oceanID string) ([]OceanLaunchSpec, error)
+}
+
+// Client is the implementation of a spot.io HTTP client.
+type Client struct {
+	baseURL   string
+	accountID string
+	client    *http.Client
```

Comment:
Looks like we could also use their golang-sdk which might make it easier and safer to call their API.

https://github.com/spotinst/spotinst-sdk-go/blob/master/service/ocean/providers/aws/service.go#L14

Reply:
Good point. I didn't even check if they had one, but seems easy to use and the http client have timeouts etc. :) I will try this one.